src/cdomains/valueDomain.ml

```diff
@@ -148,7 +148,8 @@ struct
     match t with
     | t when is_mutex_type t -> `Top
     | TInt (ik,_) -> `Int (ID.top_of ik)
-    | TFloat (FDouble, _) -> `Float (FD.top ())
+    | TFloat (FFloat as fkind, _)
+    | TFloat (FDouble as fkind, _) -> `Float (FD.top_of fkind)
```

Comment:
I think you could do 
`| TFloat ( (FFloat | FDouble) as fkind , _) -> 'Float (FD.top_of fkind)`
But you decide if that is even an improvement
(If you do, of course do it for the other cases below as well: line 170, 201, 380)